src/components/integrated/OakLessonNavItem/OakLessonNavItem.tsx

```diff
@@ -0,0 +1,304 @@
+    default:
+      return null;
+  }
+}
+
+function pickIconForSection(sectionName: LessonSectionName): OakIconName {
+  switch (sectionName) {
+    case "intro":
+      return "intro";
+    case "starter-quiz":
+    case "exit-quiz":
+      return "quiz";
+    case "video":
+      return "video";
+  }
+}
+
+function pickColorsForSection(
+  sectionName: LessonSectionName,
+): [
+  notStartedBackgroundColor: OakCombinedColorToken,
+  backgroundColor: OakCombinedColorToken,
+  borderColor: OakCombinedColorToken,
+] {
+  switch (sectionName) {
+    case "intro":
+      return [
+        "bg-decorative2-very-subdued",
+        "bg-decorative2-main",
+        "border-decorative2-stronger",
+      ];
+    case "starter-quiz":
+      return [
+        "bg-decorative1-very-subdued",
+        "bg-decorative1-main",
+        "border-decorative1-stronger",
+      ];
+    case "video":
+      return [
+        "bg-decorative4-very-subdued",
+        "bg-decorative4-main",
+        "border-decorative4-stronger",
+      ];
+    case "exit-quiz":
+      return [
+        "bg-decorative5-very-subdued",
+        "bg-decorative5-main",
+        "border-decorative5-stronger",
+      ];
+  }
+}
+
+function pickLabelForSection(sectionName: LessonSectionName): string {
+  switch (sectionName) {
+    case "intro":
+      return "Intro";
+    case "starter-quiz":
+      return "Starter quiz";
+    case "video":
+      return "Watch video";
+    case "exit-quiz":
+      return "Exit quiz";
+  }
+}
+
+function pickSummaryForProgress(props: SectionProps) {
+  switch (props.progress) {
+    case "not-started":
+      return pickSummaryForNotStarted(props);
+    case "in-progress":
+      return "In progress...";
+    case "complete":
+      return (
+        <OakFlex $gap="space-between-sssx" $alignItems="center">
+          <OakIcon
+            iconName="tick"
+            $width="all-spacing-6"
+            $height="all-spacing-6"
+          />
+          {pickSummaryForComplete(props)}
+        </OakFlex>
+      );
+  }
+}
+
+function pickSummaryForNotStarted(props: SectionProps) {
+  switch (props.lessonSectionName) {
+    case "intro":
+      return "Get ready";
+    case "starter-quiz":
+      return `${props.numQuestions} Questions`;
+    case "exit-quiz":
+      return `Practice ${props.numQuestions} questions`;
+    case "video":
+      return `${props.videoLength} min`;
+  }
+}
+
+function pickSummaryForComplete(props: SectionProps) {
+  switch (props.lessonSectionName) {
+    case "intro":
+    case "video":
+      return "Completed";
+    case "starter-quiz":
+    case "exit-quiz":
+      // The counter is rendered as the summary next
+      // when on a mobile device, so it is hidden on larger screens
+      return (
+        <>
+          <OakBox $display={["block", "none"]}>Completed</OakBox>
+          <OakBox $display={["none", "block"]}>
+            {props.grade}/{props.numQuestions} correct
+          </OakBox>
+        </>
+      );
+  }
+}
```

Comment:
I was in two minds as to whether the logic to pick the content belonged in here. I was already doing this to pick the icon so this didn't feel like a huge leap. My aim was to have something that could be readily hooked up to the relevant state in OWA.

If it wasn't here we'd have to lift all this into OWA, perhaps that is the right thing to do? Am I straying from the path?

Reply:
Yeah  I can imagine that this component is going to get quite silly at some point as we introduce more sections to lessons. At that point we can just refactor out a generalised nav component as you suggest below and then move this component over to OWA